client/asset/btc/btc.go

```diff
@@ -3719,7 +3719,7 @@ func (btc *intermediaryWallet) FindRedemption(ctx context.Context, coinID, _ dex
 		return nil, nil, fmt.Errorf("queueFindRedemptionRequest error for redemption %s: %w", outPt, err)
 	}
 
-	go btc.tryRedemptionRequests(ctx, nil, []*findRedemptionReq{req})
+	btc.tryRedemptionRequests(ctx, nil, []*findRedemptionReq{req})
```

Comment:
There seems to be no need to run this as a separate go-routine (and dcr wallet doesn't) because we are waiting on the result below anyway.

Reply:
One other thing of note here is that we set `startBlock` as `nil` here, which prompts us to find lowest height that we want to start with like this:
```
		iHeight = math.MaxInt32
		for _, req := range undiscovered {
			if req.blockHash != nil && req.blockHeight < iHeight {
				iHeight = req.blockHeight
				iHash = req.blockHash
			}
		}
```
and I'm not entirely sure that this way ^ we can't end up in a situation where `iHeight` will be == `0` which will force us to search through all blocks starting with 0th while we should have searched mempool only for this case instead, which is what DCR wallet does:
```
// Run initial search for redemption. If this contract is unmined,
// only scan mempool transactions as mempool contracts can only be
// spent by another mempool tx. If the contract is mined, scan all
// mined tx inputs starting from the block in which the contract is
// mined, up till the current best block. If the redemption is not
// found in that block range, proceed to check mempool.
if contractBlock == nil {
    dcr.findRedemptionsInMempool([]outPoint{contractOutpoint})
} else {
    dcr.tipMtx.RLock()
    bestBlock := dcr.currentTip
    dcr.tipMtx.RUnlock()
    dcr.findRedemptionsInBlockRange(contractBlock.height, bestBlock.height, []outPoint{contractOutpoint})
}
```
do we need to do something like this ^ for BTC also ?

Reply:
moved here: https://github.com/decred/dcrdex/pull/2227/files#r1138098479